### PR TITLE
Multi arch debian base-os

### PR DESCRIPTION
**Pull request by alok123t**
This PR creates a debian base-os for architectures amd64, arm64 and riscv64 using docker buildx

### CODE REVIEW

File: docs/buildx.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Running Buildx
+
+Docker Buildx is a CLI plugin that extends the docker command with the full support of the features provided by Moby BuildKit builder toolkit. BuildKit is designed to work well for building for multiple platforms and not only for the architecture and operating system that the user invoking the build happens to run. More information can be found [docker-buildx](https://docs.docker.com/buildx/working-with-buildx/)
+
+### Enable docker experimental client flag
```
Comment (pradeepk5) [inline docs/buildx.md:5]: Could you please double-check this section. 
When I followed these guidelines, the docker experimental was not enabled. 
So, I managed to work around this by explicitly specifying the config file using the following command
`docker --config ~/.docker/`
I think this is due to `$DOCKER_CONFIG` is not set by default in some installations.

File: docs/buildx.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+For more information check [docker-github](https://github.com/docker/buildx/blob/master/README.md)
+
+### Using buildx
+Docker login to enable push of images onto [dockerhub](https://hub.docker.com/u/cloudsuite/). 
+```
+docker login
+```
+
+#### Creating the base-os image
+Run `build.sh` script in `cloudsuite/commons/base-os` to generate base images for different architectures in one manifest
```
Comment (pradeepk5) [inline docs/buildx.md:51]: When I building the images with the script, I get error for riscv image. 
here is the details about the error.
```
#7 79.63 W: Failure trying to run: chroot "/rootfs" /bin/true
#7 79.63 W: See /rootfs/debootstrap/debootstrap.log for details

failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c set -ex     && echo "deb http://ftp.ports.debian.org/debian-ports sid main" >> /etc/apt/sources.list     && apt-get install -y debian-ports-archive-keyring     && dpkg --add-architecture riscv64     && apt-get update     && debootstrap     --keyring=/usr/share/keyrings/debian-ports-archive-keyring.gpg     --arch=riscv64     --variant=minbase     --include=debian-ports-archive-keyring     sid /rootfs http://ftp.ports.debian.org/debian-ports     && rm -rf /rootfs/var/log/dpkg.log /rootfs/var/log/bootstrap.log /rootfs/var/log/alternatives.log     && rm -rf /rootfs/var/cache/ldconfig/aux-cache /rootfs/var/cache/apt/*      && rm -rf /rootfs/var/lib/apt/lists/*]: buildkit-runc did not terminate successfully
```

File: docs/buildx.md (hunk 1 above)
Comment (alok123t) [reply docs/buildx.md:5]: I have specified the instructions mentioned here https://docs.docker.com/engine/reference/commandline/cli/#experimental-features
and installed docker using this method
https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script

I have only tested the above method on Ubuntu OS and seems to work fine. As you have mentioned it's not working in some installations, should we add another helpful link/instructions?

File: docs/buildx.md (hunk 2 above)
Comment (alok123t) [reply docs/buildx.md:52]: I tried to re-run on a new machine (ubuntu os, virtual machine), and the following command resolved the issue
```
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
```
I have already added this as part of readme for buildx

File: docs/buildx.md (hunk 1 above)
Comment (pradeepk5) [reply docs/buildx.md:4]: I used centOS 7 for building this image. Docker was installed from the standard repository 
```
sudo apt-get install docker-ce docker-ce-cli containerd.io
```
Also, I uninstalled the docker an installed docker again with the convenience-script you mentioned. Now, I could able to enable the experimental features of the client as you mentioned.
This is the example I followed to enable the experimental feature in a standard installation.
https://docs.docker.com/engine/swarm/configs/#simple-example-get-started-with-configs

**Pull request merged**